eb-api/src/main/java/io/engineblock/activityimpl/input/TargetRateInput.java

```diff
@@ -112,25 +112,24 @@ public void onActivityDefUpdate(ActivityDef activityDef) {
     }
 
     private void updateRateLimiter(ActivityDef activityDef) {
-        double rate = activityDef.getParams().getOptionalDoubleUnitCount("targetrate").orElse(Double.NaN);
-        if (!Double.isNaN(rate)) {
+        activityDef.getParams().getOptionalDoubleUnitCount("targetrate").ifPresent(
+        rate -> {
             if (rateLimiter==null) {
                 rateLimiter = RateLimiter.create(rate);
             } else {
+                double oldRate = rateLimiter.getRate();
                 rateLimiter.setRate(rate);
-            }
-
-            Gauge<Double> rateGauge = new Gauge<Double>() {
-                @Override
-                public Double getValue() {
-                    return rateLimiter.getRate();
+                double newRate = rateLimiter.getRate();
+                // RateLimiter turns 30000.0 into 29999.999999999996 so we check if the internal value changed here
```

Comment:
This comment seems out of place, and thus confusing. Should it be removed or changed?

Reply:
The comment explains why we can not compare `rateLimiter.getRate()` to the `rate` parameter we will use in `rateLimiter.setRate(rate);` since the `RateLimiter` class modifies the actually saved value, i.e. turns `30000.0` into `29999.99...6`.
I found this behavior surprising enough, to explain in a comment why we instead compare `rateLimiter.getRate()` before and after calling `setRate`.

If you think this is clear to the reader and no-one will be inclined to "optimize" this check in the future, feel free to reword or remove the comment.